appservice/src/tree/FileTreeItem.ts

```diff
@@ -3,25 +3,28 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AzureParentTreeItem, AzureTreeItem, openReadOnlyContent, TreeItemIconPath } from 'vscode-azureextensionui';
+import { AzExtParentTreeItem, AzExtTreeItem, openReadOnlyContent, TreeItemIconPath } from 'vscode-azureextensionui';
 import { ext } from '../extensionVariables';
 import { getFile, IFileResult } from '../getFile';
+import { IFilesClient } from '../IFilesClient';
 import { localize } from '../localize';
 import { getThemedIconPath } from './IconPath';
-import { ISiteTreeRoot } from './ISiteTreeRoot';
 
 /**
  * NOTE: This leverages a command with id `ext.prefix + '.openFile'` that should be registered by each extension
  */
-export class FileTreeItem extends AzureTreeItem<ISiteTreeRoot> {
+export class FileTreeItem extends AzExtTreeItem {
     public static contextValue: string = 'file';
     public readonly contextValue: string = FileTreeItem.contextValue;
     public readonly label: string;
     public readonly path: string;
     public readonly isReadOnly: boolean;
 
-    constructor(parent: AzureParentTreeItem, label: string, path: string, isReadOnly: boolean) {
+    public readonly client: IFilesClient;
+
+    constructor(parent: AzExtParentTreeItem, client: IFilesClient, label: string, path: string, isReadOnly: boolean) {
```

Comment:
> I think looks good more or less! I had a couple of questions, but I don't think that they're necessarily blocking.
> 
> 1. Since we're relying on SiteClient having all of the functions/properties needed by IAppSettingsClient and IFilesClient, I think we should say that the SiteClient implements both of these clients so it's more obvious when we just pass SiteClient as the IAppSettingsClient in the actual extension repos.
> 2. As far as I know, everything that uses the Files/Log Files also uses Kudu.  But do we want to extract some of the kudu code out so we can make it more generic?  For example, I don't know if we'd be able to use this for SWA (or if we'd even want to for that matter).  But I'm pretty sure in the SWA case, we'd have to make API calls to GitHub to retrieve the vfs, etc.

For 1: I made it implement IFilesClient, will make it implement IAppSettingsClient in a new (extremely small) PR.
For 2: Is it okay if we leave this change for when we do decide if SWA is able to use this?